lms/resources/_js_config.py

```diff
@@ -75,6 +75,19 @@ def add_document_url(self, document_url):
             self._config["viaUrl"] = via_url(self._request, document_url)
             self._add_canvas_speedgrader_settings(document_url=document_url)
 
+    def add_vitalsource_launch_config(self, book_id, cfi=None):
+        vitalsource_svc = self._request.find_service(name="vitalsource")
+        launch_url, launch_params = vitalsource_svc.get_launch_params(
+            book_id, cfi, self._request.lti_user
+        )
+        self._config["vitalSource"] = {
+            "launchUrl": launch_url,
+            "launchParams": launch_params,
+        }
+        self._add_canvas_speedgrader_settings(
+            vitalsource_book_id=book_id, vitalsource_cfi=cfi
+        )
```

Comment:
I'm filling out these SpeedGrader params here in a way that is consistent with how Canvas files launches are handled. However I haven't actually tested the VS viewer with SpeedGrader yet. We could just leave that out of scope for this PR.